base/abstractarray.jl

```diff
@@ -343,6 +343,34 @@ function first(itr)
     x[1]
 end
 
+"""
+    first(predicate, coll)
+
+Get the first element of `coll` satisfying `predicate` wrapped in `Some`.
+
+If no element of `coll` satisfies `predicate`, return `nothing`.
+
+# Examples
+```jldoctest
+julia> first(>(5), 1:10)
+Some(6)
+
+julia> first(isodd, 2:2:10) |> isnothing
+true
+
+julia> first(iseven, [5; 3; 4; 2; 6; 8]) |> something
+4
+
+julia> something(first(>(10), 1:10), 0)
+0
+```
+"""
+function first(predicate, itr)
+    for x in itr
+        predicate(x) && return Some(x)
+    end
```

Comment:
```suggestion
    end
    return nothing
```

Reply:
Is there a style guide for Julia base somewhere? I didn't know it was preferred to always use `return`, and I don't see it listed in https://docs.julialang.org/en/v1/manual/style-guide/

Reply:
It's just a bit iffy to rely on the fact that `for` loops return `nothing`, so it's better to be explicit in the case that the loop completes.